packages/owl-bot/src/scan-and-retry-failed-lock-updates.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {google} from '@google-cloud/cloudbuild/build/protos/protos';
+import {core} from './core';
+
+const twentyFourHours =
+  60 * // seconds
+  60 * // minutes
+  24; // hours.
+
+export async function scanAndRetryFailedLockUpdates(
+  projectId: string,
+  triggerId: string,
+  maxTries: number
+): Promise<void> {
+  const cb = core.getCloudBuildInstance();
+  const builds = cb.listBuildsAsync({projectId, pageSize: 100});
+  const rebuilds = await filterBuildsToRetry(triggerId, maxTries, builds);
+  console.info(`${rebuilds.length} to rebuild`);
+  for (const build of rebuilds) {
+    console.info(`Triggering rebuild for ${build.name}`);
+    console.debug(build);
+    await cb.retryBuild({
+      name: build.name,
+      projectId: build.projectId,
+      id: build.id,
+    });
+  }
+}
+
+/**
+ * Scans all the cloud builds for the past 24 hours looking for failures for
+ * the given build trigger.  Returns a list of builds that should be retried
+ * because they never succeeded.
+ */
+export async function filterBuildsToRetry(
+  triggerId: string,
+  maxTries: number,
```

Comment:
what is maxTries? Should it be a constant?

Reply:
Added comment.